cirq-google/cirq_google/engine/engine_job.py

```diff
@@ -340,19 +343,42 @@ def calibration_results(self) -> Sequence[CalibrationResult]:
             self._calibration_results = cal_results
         return self._calibration_results
 
-    @classmethod
-    def _get_batch_results_v2(
-        cls, results: v2.batch_pb2.BatchResult
-    ) -> Sequence[Sequence[cirq.Result]]:
+    def _get_job_results_v1(self, result: v1.program_pb2.Result) -> Sequence[EngineResult]:
+        # coverage: ignore
+        job_id = self.id()
+        job_finished = self.update_time()
+
         trial_results = []
-        for result in results.results:
-            # Add a new list for the result
-            trial_results.append(_get_job_results_v2(result))
+        for sweep_result in result.sweep_results:
+            sweep_repetitions = sweep_result.repetitions
+            key_sizes = [(m.key, len(m.qubits)) for m in sweep_result.measurement_keys]
+            for result in sweep_result.parameterized_results:
+                data = result.measurement_results
+                measurements = v1.unpack_results(data, sweep_repetitions, key_sizes)
+
+                trial_results.append(
+                    EngineResult(
+                        params=cirq.ParamResolver(result.params.assignments),
+                        measurements=measurements,
+                        job_id=job_id,
+                        job_finished_time=job_finished,
+                    )
+                )
         return trial_results
 
-    @classmethod
-    def _flatten(cls, result) -> Sequence[cirq.Result]:
-        return [res for result_list in result for res in result_list]
+    def _get_job_results_v2(self, result: v2.result_pb2.Result) -> Sequence[EngineResult]:
+        sweep_results = v2.results_from_proto(result)
+        # Flatten to single list to match to sampler api.
+        return [
+            EngineResult.from_result(result, job_id=self.id(), job_finished_time=self.update_time())
```

Comment:
nit: grab `self.id()` and `self.update_time()` once and then pass them in to each `from_result` call, as in `_get_job_results_v1` above.

Reply:
done